temporian/core/test/serialization_test.py

```diff
@@ -163,109 +175,269 @@ def test_serialize_attributes(self):
+            timestamps=[1.0, 2.0, 3.0],
+            features={"f1": [100.0, 200.0, 300.0]},
+        )
+        y = tp.event_set(
+            timestamps=[1.0, 2.0, 3.0],
+            features={"f2": [1.0, 2.0, 3.0]},
+            same_sampling_as=x,
+        )
+        z = tp.event_set(
+            timestamps=[1.0, 2.0, 3.0],
+            features={"f3": [3.0, 5.0, 2.0]},
+            same_sampling_as=x,
+        )
+
+        result = f(x, y, z)
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            path = os.path.join(tempdir, "my_fn.tem")
+            tp.save(f, path, x=x, y=y, z=z)
+            loaded_f = tp.load(path=path)
+
+        loaded_result = loaded_f(x=x, y=y, z=z)
+
+        self.assertEqual(result, loaded_result)
+
+    def test_load_args_and_kwargs(self):
+        @tp.compile
+        def f(x: EventSetNode, y: EventSetNode, z: EventSetNode):
+            return {"output": tp.glue(x, y, z)}
 
         x = tp.event_set(
             timestamps=[1.0, 2.0, 3.0],
             features={"f1": [100.0, 200.0, 300.0]},
         )
-        results = f(x)
+        y = tp.event_set(
+            timestamps=[1.0, 2.0, 3.0],
+            features={"f2": [1.0, 2.0, 3.0]},
+            same_sampling_as=x,
+        )
+        z = tp.event_set(
+            timestamps=[1.0, 2.0, 3.0],
+            features={"f3": [3.0, 5.0, 2.0]},
+            same_sampling_as=x,
+        )
+
+        result = f(x, y, z)
 
         with tempfile.TemporaryDirectory() as tempdir:
             path = os.path.join(tempdir, "my_fn.tem")
-            tp.save(f, inputs={"x": x}, path=path)
-            input, outputs = tp.load(path=path, squeeze=True)
+            tp.save(f, path, x=x, y=y, z=z)
+            loaded_f = tp.load(path=path)
+
+        loaded_result = loaded_f(x, y, z)
 
-        self.assertEqual(list(outputs.keys()), ["a", "b"])
+        self.assertEqual(result, loaded_result)
+
+    def test_load_signature(self):
+        """Checks that help(loaded_f) shows the correct param spec."""
 
-        loaded_results = tp.run(outputs, {input: x})
+        @tp.compile
+        def f(x: EventSetNode, y: EventSetNode, z: EventSetNode):
+            return {"output": tp.prefix("a_", x)}
 
-        self.assertEqual(results["a"], loaded_results["a"])
-        self.assertEqual(results["b"], loaded_results["b"])
+        with tempfile.TemporaryDirectory() as tempdir:
+            path = os.path.join(tempdir, "my_fn.tem")
+            tp.save(f, path, x=self.evset, y=self.evset, z=self.evset)
+            loaded_f = tp.load(path=path)
+
+        doc = pydoc.render_doc(loaded_f)
+        self.assertTrue(
+            "x: temporian.core.data.node.EventSetNode, "
+            "y: temporian.core.data.node.EventSetNode, "
+            "z: temporian.core.data.node.EventSetNode"
+            in doc
+        )
+        self.assertTrue(
+            "-> Dict[str, temporian.core.data.node.EventSetNode]" in doc,
+        )
 
 
 if __name__ == "__main__":

```

Comment:
Can you also the the following scenarios:

Function compositions
===================

@tp.compile
def f(x: tp.EventSetOrNode):
    return tp.prefix("f_",x)
    
@tp.compile
def g(x: tp.EventSetOrNode):
    return tp.prefix("g_",x)
    
@tp.compile
def h(x: tp.EventSetOrNode):
    return f(g(x))
    
tp.save(f, ...)

Partial function
=============

@tp.compile
def f(x: tp.EventSetOrNode, b: bool):
    if b:
        return tp.prefix("true_",x)
    else:
        return tp.prefix("false_",x)
    
tp.save(partial(f,b=True), ...)

and

Recursive function
==============

@tp.compile
def f(x: tp.EventSetOrNode, n: int=5):
    if n>0:
        return f(tp.prefix(f"{n}_",x), n=n-1)
    else:
        return x
    
tp.save(... # Not sure if you need to have a partial here.

Reply:
great suggestions! done